build.rs

```diff
@@ -0,0 +1,18 @@
+const TEST_EXPR: &str = "{
+#[cfg(unix)]
+fn _test<I: std::os::unix::io::AsFd>(_: I) {}
+#[cfg(unix)]
+fn _test2(_: std::os::unix::io::OwnedFd) {}
+#[cfg(windows)]
+fn _test<I: std::os::windows::io::AsSocket>(_: I) {}
+#[cfg(windows)]
+fn _test2(_: std::os::windows::io::OwnedSocket) {}
+}";
+
+fn main() {
+    let cfg = autocfg::new();
+
+    if cfg.probe_expression(TEST_EXPR) {
+        autocfg::emit("has_io_safety");
+    }
```

Comment:
I would prefer to invert cfg here. See [slab's build script](https://github.com/tokio-rs/slab/blob/master/build.rs#L15-L20) for example and reason.

Reply:
Also, calling rustc in a build script to compile code is relatively fragile (does not work in some host environments and does not work on tier 3 targets). So I would recommend checking the version.

If you need older nightly support, you can consider [this](https://github.com/tokio-rs/tokio/blob/master/tokio/build.rs#L26-L44) approach.